app/overrides/add_activation_keys_input.rb

```diff
@@ -10,12 +10,12 @@
 
 Deface::Override.new(:virtual_path => "hostgroups/_form",
                      :name => "hostgroups_update_environments_select",
-                     :replace => 'erb[loud]:contains("select_f"):contains(":environment_id")',
+                     :insert_bottom => 'div#primary',
```

Comment:
I think it makes more sense for content view and lifecycle environment to be just prior to the puppet environment selector, as your selection for those values can change the value of the puppet env